pkg/codegen/docs/gen.go

```diff
@@ -2016,33 +2017,38 @@ func (dctx *docGenContext) getMod(
 	add bool,
 ) *modContext {
 	modName := pkg.TokenToModule(token)
-	mod, ok := modules[modName]
+	return dctx.getModByModName(pkg, tokens.ModuleName(modName), tokenPkg, modules, tool, add)
+}
+
+func (dctx *docGenContext) getModByModName(
```

Comment:
@tmeckel when reviewing your PR noticed that getMod() recurses indefinitely for google-native. How about something like this change.. we get it recurse over ModuleName instead of token, and try to get parent computation more precise.

Reply:
@t0yv0 just came up with another solution of my own. Let me check yours after mine. But when your solution works as well, we should adopt yours because I had to introduce `HasModuleFormat` on the `type Package struct`

Reply:
Ah very neat. Well I'll let you drive and check your PR again tomorrow. Thanks.!